src/atomWithHash.ts

```diff
@@ -34,7 +34,7 @@ export function atomWithHash<Value>(
   if (setHashOption === 'replaceState') {
     setHash = (searchParams) => {
       window.history.replaceState(
-        null,
+        window.history.state,
```

Comment:
Nice catch. I didn't realize this would be better.
ref: https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState